### PR TITLE
[nest] Remove the access token when the thing is removed

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #14818

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/sdm/api/PubSubAPI.java [hunk 1]
```diff
@@ -153,14 +155,21 @@ public PubSubAPI(HttpClientFactory httpClientFactory, OAuthFactory oAuthFactory,
...
         scheduler = new ScheduledThreadPoolExecutor(3, new NamedThreadFactory(ownerId, true));
     }
 
     public void dispose() {
         subscriptionListeners.clear();
         scheduler.shutdownNow();
+        oAuthFactory.ungetOAuthService(oAuthServiceHandleId);
+    }
+
+    public void deleteServiceAndAccessToken() {
```
Comment (jlaur) [inline bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/sdm/api/PubSubAPI.java:171]: Perhaps to add some context:
```suggestion
    public void deleteOAuthServiceAndAccessToken() {
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Otherwise LGTM. I'll let @wborn have a look also.

File: bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/sdm/api/PubSubAPI.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/sdm/api/PubSubAPI.java]: Method renamed

**Review by jlaur — APPROVED** [review]

**Review by wborn — APPROVED** [review]
LGTM!

**Pull request merged**